### PR TITLE
EFA-605 Workaround for messages with no from header

### PR DESCRIPTION
**Pull request by ig-garcia**

### CODE REVIEW

File: k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java [hunk 1]
```diff
@@ -317,14 +317,40 @@ public void setSubject(String subject) {
     public Address[] getFrom() {
         if (mFrom == null) {
             String list = MimeUtility.unfold(getFirstHeader("From"));
-            if (list == null || list.length() == 0) {
+            if (list == null || list.isEmpty()) {
                 list = MimeUtility.unfold(getFirstHeader("Sender"));
             }
+            if (list == null || list.isEmpty()) {
```
Comment (Stereodark) [inline k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java:323]: do we need to check the same condition twice? The line below can be in the IF above, right?

**Review by Stereodark — APPROVED** [review]
looks good to me, just minor optimization, an if is checked twice for the same condition

File: k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java (hunk 1 above)
Comment (ig-garcia) [reply k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java:323]: The thing is that list can still be null or empty after calling the MimeUtility.unfold. So we need the check. If it does not look good then maybe some function can be extracted.

Comment (Stereodark) [reply k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java:323]: right...silly me, still half sleeping this morning, of course, forget the comment 👍

Comment (ig-garcia) [reply k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java]: Please check now, I improved the logic to avoid repetition.

Comment (Stereodark) [reply k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java]: very nice, so also future headers check are also possible by adding to the list. Cool

**Review by Stereodark — APPROVED** [review]

**Pull request merged**